src/Nethermind/Nethermind.Consensus/Producers/TxPoolTxSource.cs

```diff
@@ -77,52 +122,81 @@ public IEnumerable<Transaction> GetTransactions(BlockHeader parent, long gasLimi
                 bool success = _txFilterPipeline.Execute(tx, parent);
                 if (!success) continue;
 
-                if (tx.SupportsBlobs)
+                foreach (Transaction blobTx in PickBlobTxsBetterThanCurrentTx(selectedBlobTxs, tx, comparer))
```

Comment:
I'm wondering if we can do it in more efficient/simpler way - let's discuss it.

Reply:
It's already simplified a lot comparing to initial version 😄 But I'm open for discussion. What is important here, we need to have all transactions (mix of blob and non-blob) in correct order, so we can't add blobs at the beginning or at the end. 
In current implementation we are picking max possible number of blob transactions which can be included in the block (from Eip4844Constants - 6 blobs for now) and then we are iterating non-blob txs and adding blob txs in between according to priority fee, from highest to lowest